api/odl.py

```diff
@@ -280,19 +287,26 @@ def checkout(self, patron, pin, licensepool, internal_format):
         if loan.count() > 0:
             raise AlreadyCheckedOut()
 
-        doc = self.get_license_status_document(licensepool, patron)
+        # Create a local loan so it's database id can be used to
+        # receive notifications from the distributor.
+        loan, ignore = get_one_or_create(_db, Loan, patron=patron, license_pool_id=licensepool.id)
+
+        doc = self.get_license_status_document(loan)
         status = doc.get("status")
 
         if status not in [self.READY_STATUS, self.ACTIVE_STATUS]:
             # Something went wrong with this loan and we don't actually
             # have the book checked out. This should never happen.
+            # Remove the loan we created.
+            _db.delete(loan)
```

Comment:
If there's a preexisting loan, and then `checkout` is called, and it's revealed that the status is not 'ready' or 'active', then the loan will be deleted. I think this is correct, or at least not a problem, but I want to confirm because it's not what I was expecting to happen here.

Reply:
If there's a preexisting loan, checkout will raise AlreadyCheckedOut() before it gets here.